keg_elements/db/mixins.py

```diff
@@ -67,30 +67,116 @@ def from_dict(self, data):
             else:
                 setattr(self, key, value)
 
-    def to_dict(self, exclude=None):
-        if exclude is None:
-            exclude = []
-        insp = sainsp(self)
-        return dict((attr.key, attr.value)
-                    for attr in insp.attrs if attr.key not in exclude)
+    def to_dict(self, exclude=[], hybrids=[]):
+        """Covert the object properties to a dictionary.
+
+        :param exclude: a list of columns to ignore
+        :param hybrids: a list of the hybrid properties to include in the dictionary.
+        :returns: a dictionary representation of the object
+
+        .. note: By default hybrid properties are not included in the returned dict. To add a hybrid
+        property to the returned dict pass a list of the property names and they will be included.
+        """
+        data = dict([(name, getattr(self, name))
+                     for name in self.column_names()
+                     if name not in exclude])
+
+        for hybrid in hybrids:
+            data[hybrid] = getattr(self, hybrid)
+
+        return data
+
+    @might_commit
+    @might_flush
+    @classmethod
+    def add(cls, **kwargs):
+        obj = cls()
+        obj.from_dict(kwargs)
+        db.session.add(obj)
+        return obj
+
+    @might_commit
+    @might_flush
+    @classmethod
+    def delete(cls, oid):
+        """Delete an object from the session
+
+        :param oid: the object identifier, normally the primary key
+        :rtype: bool
+        :return: The result of the operation
+        """
+        obj = cls.query.get(oid)
+
+        if obj is None:
+            return False
+
+        db.session.delete(obj)
+        return True
+
+    @might_commit
+    @classmethod
+    def delete_cascaded(cls):
+        cls.query.delete(synchronize_session=False)
+        db.session.expire_all()
+
+    @classmethod
+    def column_names(cls):
+        return {col.key for col in cls.__mapper__.columns}
```

Comment:
I never really thought about getting rid of columns from the collection like that.  I like it, although wondering how often I would use it.  What if we made the return type optional:

`def column_names(cls, as_set=False):` ?